app/code/Magento/AdminNotification/Controller/Adminhtml/Notification/AjaxMarkAsRead.php

```diff
@@ -6,12 +6,35 @@
  */
 namespace Magento\AdminNotification\Controller\Adminhtml\Notification;
 
+use Magento\Backend\App\Action;
+use Magento\Framework\Controller\ResultFactory;
+
 class AjaxMarkAsRead extends \Magento\AdminNotification\Controller\Adminhtml\Notification
 {
     /**
+     * @var \Magento\AdminNotification\Model\NotificationService
+     */
+    private $notificationService;
+
+    /**
+     * @param Action\Context $context
+     * @param \Magento\AdminNotification\Model\NotificationService|null $notificationService
+     * @throws \RuntimeException
+     */
+    public function __construct(
+        Action\Context $context,
+        \Magento\AdminNotification\Model\NotificationService $notificationService = null
+    ) {
+        parent::__construct($context);
+        $this->notificationService = $notificationService?: \Magento\Framework\App\ObjectManager::getInstance()
+            ->get(\Magento\AdminNotification\Model\NotificationService::class);
+    }
+
+    /**
      * Mark notification as read (AJAX action)
      *
-     * @return void
+     * @return \Magento\Framework\Controller\Result\Json|void
```

Comment:
I think it should follow the `\Magento\Framework\App\ActionInterface` interface signature, since it implements it.